packages/payload/src/queues/operations/local/index.ts

```diff
@@ -0,0 +1,63 @@
+import type { RunningJobFromTask } from '../../config/types/workflowTypes.js'
+
+import {
+  createLocalReq,
+  type Payload,
+  type PayloadRequest,
+  type RunningJob,
+  type TypedJobs,
+} from '../../../index.js'
+import { runAllJobs } from '../runAllJobs/index.js'
+
+export const getJobsLocalAPI = (payload: Payload) => ({
+  queue: async <
+    // eslint-disable-next-line @typescript-eslint/no-duplicate-type-constituents
+    TTaskOrWorkflowSlug extends keyof TypedJobs['tasks'] | keyof TypedJobs['workflows'],
+  >(
+    args:
+      | {
+          input: TypedJobs['tasks'][TTaskOrWorkflowSlug]['input']
+          // TTaskOrWorkflowlug with keyof TypedJobs['workflows'] removed:
+          task: TTaskOrWorkflowSlug extends keyof TypedJobs['tasks'] ? TTaskOrWorkflowSlug : never
+          workflow?: never
+        }
+      | {
+          input: TypedJobs['workflows'][TTaskOrWorkflowSlug]['input']
+          task?: never
+          workflow: TTaskOrWorkflowSlug extends keyof TypedJobs['workflows']
+            ? TTaskOrWorkflowSlug
+            : never
+        },
+  ): Promise<
+    TTaskOrWorkflowSlug extends keyof TypedJobs['workflows']
+      ? RunningJob<TTaskOrWorkflowSlug>
+      : RunningJobFromTask<TTaskOrWorkflowSlug>
+  > => {
+    return (await payload.create({
```

Comment:
We need to be able to pass the `req` to use the same transaction if you're calling from a hook.

Reply:
Done